src/openms/source/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.cpp

```diff
@@ -0,0 +1,654 @@
+               "IsotopeCosine\tChargeIntensityCosine\tMassSNR\tMaxQScoreCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\n";
+
+      }
+      else
+      {
+        fs
+            << "MassIndex\tSpecIndex\tFileName\tScanNum\tRetentionTime\tMassCountInSpec\tAverageMass\tMonoisotopicMass\t"
+               "SumIntensity\tMinCharge\tMaxCharge\t"
+               "PeakCount\t"
+               "PrecursorScanNum\tPrecursorMz\tPrecursorIntensity\tPrecursorCharge\tPrecursorMonoisotopicMass\tPrecursorQScore\t"
+               "IsotopeCosine\tMassSNR\tMaxQScoreCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\n";
+      }
+    }
+  }
+
+  void DeconvolutedSpectrum::writeAttCsvHeader(std::fstream &fs)
+  {
+    fs
+        << "ScanNumber,RetentionTime,PrecursorScanNumber,Charge,ChargeSNR,PeakIntensity,EnvIntensity,EnvIsotopeCosine,PeakMz,"
+           "MonoMass,MassSNR,IsotopeCosine,ChargeIntensityCosine,MassIntensity,QScore,Class\n";
+  }
+
+  void DeconvolutedSpectrum::writeThermoInclusionHeader(std::fstream &fs)
+  {
+    fs << "Compound,Formula,Adduct,m/z,z,t start (min),t stop (min),Isolation Window (m/z),Normalized AGC Target (%)\n";
+  }
+
+
+  void DeconvolutedSpectrum::clearPeakGroupsChargeInfo()
+  {
+    for (auto &pg: peakGroups)
+    {
+      pg.clearChargeInfo();
+    }
+  }
+
+  void DeconvolutedSpectrum::writeAttCsv(std::fstream &fs,
+                                         int msLevel,
+                                         double qScoreThreshold = -1000,
+                                         int numMaxMS2 = -1)
+  {
+
+    if (msLevel > 1)
+    {
+      if (precursorPeakGroup != nullptr)
+      {
+        fs << scanNumber << "," << precursorPeakGroup->spec->getRT() << "," << precursorPeakGroup->scanNumber << ","
+           << precursorPeak.getCharge()
+           << "," << log10(precursorPeakGroup->perChargeInfo[precursorPeak.getCharge()][0] + 1e-3) << ","
+           << log10(precursorPeak.getIntensity() + 1)
+           << "," << log10(precursorPeakGroup->perChargeInfo[precursorPeak.getCharge()][2] + 1)
+           << "," << precursorPeakGroup->perChargeInfo[precursorPeak.getCharge()][1]
+           << "," << precursorPeak.getMZ()
+           << "," << precursorPeakGroup->monoisotopicMass << "," << log10(precursorPeakGroup->totalSNR + 1e-3) << ","
+           << precursorPeakGroup->isotopeCosineScore
+           << "," << precursorPeakGroup->chargeCosineScore << "," << log10(precursorPeakGroup->intensity + 1)
+           << "," << precursorPeakGroup->qScore
+           << ",f\n";
+      }
+      else
+      {
+        fs << scanNumber << "," << spec->getRT() << "," << 0 << "," << 0
+           << ",?,"
+           << log10(precursorPeak.getIntensity() + 1)
+           << ",?,?," << (precursorPeak.getMZ())
+           << "," << 0 << ",?,?,?,?"
+           << "," << precursorPeakGroup->qScore
+           << ",f\n";
+      }
+    }
+    else
+    {
+      //double scoreThreshold = 0;
+      std::vector<double> scores;
```

Comment:
sorry I misread that somehow. Then it would be better to just put the scores vector inside the if block. THen the swap is not needed at all